python-sdk/src/astro/sql/operators/base_decorator.py

```diff
@@ -193,42 +193,52 @@ def translate_jinja_to_sqlalchemy_template(self, context: dict) -> None:
         if context:
             self.sql = self.render_template(self.sql, context)
 
-    def get_openlineage_facets(self, task_instance):
+    def get_openlineage_facets_on_complete(self, task_instance):
         """
         Returns the lineage data
         """
         from astro.lineage import (
             BaseFacet,
             DataSourceDatasetFacet,
             OpenlineageDataset,
+            OperatorLineage,
             OutputStatisticsOutputDatasetFacet,
             SchemaDatasetFacet,
             SchemaField,
             SqlJobFacet,
         )
-        from astro.lineage.extractor import OpenLineageFacets
 
         input_dataset: list[OpenlineageDataset] = []
         output_dataset: list[OpenlineageDataset] = []
-        if (
-            self.output_table.openlineage_emit_temp_table_event() and self.output_table.conn_id
-        ):  # pragma: no cover
+
+        first_table = find_first_table(
```

Comment:
there is a first_table in execute method, would making that a `self.first_table` help?

Reply:
`self.first_table` is not accessible in get_openlineage_facet. Hence its required

Reply:
but other class variables like `self.op_args` is accessible so can't we access this in a similar way?

Reply:
No, it's not working for other classes as well, in open lineage because of an airflow bug.